lib/ansible/runner.rb

```diff
@@ -8,13 +8,35 @@ def run(env_vars, extra_vars, playbook_path)
       private
 
       def run_via_cli(env_vars, extra_vars, playbook_path)
-        result = AwesomeSpawn.run!(ansible_command, :env => env_vars, :params => [{:extra_vars => JSON.dump(extra_vars)}, playbook_path])
-        JSON.parse(result.output)
+        Dir.mktmpdir("ansible-runner") do |base_dir|
+          mkdir(base_dir + '/project') # without this, there is a silent fail of the ansible-runner command see https://github.com/ansible/ansible-runner/issues/88
+
+          result = AwesomeSpawn.run!(ansible_command(base_dir), :env => env_vars, :params => [{:cmdline => "--extra-vars '#{JSON.dump(extra_vars)}'", :playbook => playbook_path}])
+
+          parsed_stdout = []
+
+          # output is JSON per new line
+          result.output.each_line do |line|
+            # TODO(lsmola) we can remove exception handling when this is fixed
+            # https://github.com/ansible/ansible-runner/issues/89#issuecomment-404236832 , so it fails early if there is
+            # a non json line
+            begin
+              parsed_stdout << JSON.parse(line)
+            rescue => e
+              _log.warn("Couldn't parse JSON from: #{e}")
+            end
+          end
+          parsed_stdout
+        end
+      end
+
+      def mkdir(base_dir)
+        Dir.mkdir(base_dir) unless Dir.exist?(base_dir)
```

Comment:
Can you just rescue EEXIST and save the conditional?  Saves a `stat()` system call :smile: